In [1]:
import pandas as pd
import numpy as np

In [2]:
data = pd.read_csv('webpage.csv')

In [3]:
data.head()

Visit Number  Page A  Page B  Page C
0             1    21.2     NaN     NaN
1             2     NaN     NaN    26.3
2             3     NaN    31.8     NaN
3             4     NaN     NaN    28.9
4             5    30.7     NaN     NaN

In [4]:
sample_B = data['Page B']

In [5]:
sample_B.dropna(inplace = True)
sample_B.reset_index(inplace = True, drop = True)

In [6]:
sample_B.head()

0    31.8
1    29.5
2    25.5
3    27.4
4    32.1
Name: Page B, dtype: float64

In [7]:
sample_B.describe()

count    15.000000
mean     29.926667
std       1.921483
min      25.500000
25%      28.750000
50%      30.400000
75%      31.300000
max      32.700000
Name: Page B, dtype: float64

H0 = 24 <p>
HA > 24 <p>
Page mean = 29.93, is the sample mean, not the popularity mean <p>
cannot reject the null hypothesis

In [8]:
import scipy.stats as stats

In [9]:
stats.ttest_1samp(sample_B, 24, alternative = 'greater')
#pvalue is very small, null hypothesis should be rejected

Ttest_1sampResult(statistic=11.945922250959974, pvalue=4.955711171013671e-09)

In [10]:
xbar = sample_B.mean()
sem = stats.sem(sample_B)
degree = len(sample_B) - 1
stats.t.interval(0.95, degree, xbar, sem)
#population mean of Page B

(28.862585091100897, 30.990748242232428)

In [11]:
sample_A = data['Page A']
sample_A.dropna(inplace = True)
sample_A.reset_index(inplace = True, drop = True)

In [12]:
stats.ttest_ind(sample_A, sample_B, equal_var = True, alternative = 'two-sided')
#pvalue is super small, reject the null hypothesis

Ttest_indResult(statistic=-6.991257775978399, pvalue=9.082443770609048e-08)

In [13]:
sample_C = data['Page C']
sample_C.dropna(inplace = True)
sample_C.reset_index(inplace = True, drop = True)

In [14]:
stats.f_oneway(sample_A, sample_B, sample_C)

F_onewayResult(statistic=31.561864527023303, pvalue=2.7168065130727566e-09)